feature/google/src/test/java/com/revenuecat/purchases/google/SubscriptionOptionTest.kt

```diff
@@ -0,0 +1,165 @@
+package com.revenuecat.purchases.google
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.GooglePurchasingData
+import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
+import com.revenuecat.purchases.utils.mockProductDetails
+import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SubscriptionOptionTest {
+    @Test
+    fun `SubscriptionOption can find recurring phase with INFINITE_RECURRING`() {
+        val productDetails = mockProductDetails()
+
+        val recurringPhase = PricingPhase(
+            billingPeriod = "P1M",
+            recurrenceMode = RecurrenceMode.INFINITE_RECURRING,
+            billingCycleCount = 0,
+            price = Price(
+                formatted = "$9.00",
+                amountMicros = 9000000,
+                currencyCode = "USD",
+            )
+        )
+
+        val subscriptionOption = GoogleSubscriptionOption(
+            id = "subscriptionOptionId",
+            pricingPhases = listOf(recurringPhase),
+            tags = emptyList(),
+            purchasingData = GooglePurchasingData.Subscription(
+                productId = "product_id",
+                productDetails = productDetails,
+                optionId = "subscriptionOptionId",
+                token = "mock-token"
+            )
+        )
+
+        assertThat(subscriptionOption.freePhase).isNull()
+        assertThat(subscriptionOption.introPhase).isNull()
+        assertThat(subscriptionOption.fullPricePhase).isEqualTo(recurringPhase)
+    }
+
+    @Test
+    fun `SubscriptionOption can find recurring phase with FINITE_RECURRING`() {
```

Comment:
still wondering if FINITE_RECURRING would be the base plan ever..

Reply:
It looks like prepaids are `NON_RECURRING` so I'm going to just remove the filter and just grab the last phase no matter what 🤷‍♂️